src/filesystem.ts

```diff
@@ -97,6 +101,46 @@ export async function loadFileSystem(
 }
 
 
+/**
+ * Create a filesystem
+ *
+ * @param permissions The permissions to initialize the filesystem
+ */
+export const createFilesystem = async (permissions: Permissions): Promise<FileSystem> => {
+  // Get or create root read key
+  const rootKey = await readKey()
+
+  // Create an empty filesystem
+  const fs = await FileSystem.empty({ permissions, rootKey })
+
+  // Self-authorize a filesystem UCAN
+  const issuer = await did.write()
+  const proof: string | null = await storage.getItem("ucan")
+  const fsUcan = await token.build({
+    potency: "APPEND",
+    resource: "*",
+    proof: proof ? proof : undefined,
+    lifetimeInSeconds: 60 * 60 * 24 * 30 * 12 * 1000, // 1000 years
+
+    audience: issuer,
+    issuer
+  })
+
+  // Add filesystem UCAN to store
+  await ucan.store([token.encode(fsUcan)])
+
+  // Update filesystem and publish data root
+  const rootCid = await fs.root.put()
+  const res = await dataRoot.update(rootCid, token.encode(fsUcan))
+  if (!res.success) console.warn("Failed to update data root.")
+
+  // Update CID log
+  await cidLog.add(rootCid.toString())
+
+  return fs
```

Comment:
We should also add the sample data here (see load filesystem func).

Reply:
Discussed this with @therealjeffg. The decision was to only provide `public` and `private` directories by default in app-owned apps.